torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -190,62 +194,6 @@ class ProcessGroupNCCL : public ProcessGroup {
     bool isHighPriorityStream;
   };
 
-  // FutureNCCL is a subclass of ivalue's Future. The goal is to use
-  // this class in getFuture API of WorkNCCL. This Future is mostly a
-  // wrapper to synchronize streams appropriately and it mostly enables
-  // the async programming model of CUDA while trying to adhere to the
-  // Future interface. FutureNCCL does not support NCCL_BLOCKING_WAIT flag
-  // or NCCL's barrier().
-  //
-  // If created by WorkNCCL's getFuture API, FutureNCCL has a reference to
-  // WorkNCCL's cudaEvents, NCCL collective's outputs, and the device indices of
-  // outputs' devices. Its value is NCCL collective's outputs.
-  //
-  // If created by FutureNCCL's then callback, its value becomes the value of
-  // callback() and its cudaEvents will record the NCCL stream that runs that
-  // callback. Before invoking the callback, FutureNCCL will synchronize its
-  // own cudaEvents with the stream that runs the callback. This design
-  // enables synchronizing the appropriate streams and avoids stalling PyTorch's
-  // default stream while running the callback. In case of multiple then
-  // callbacks, each will be executed on its own fresh stream.
-  struct FutureNCCL : at::cuda::CUDAFuture {
```

Comment:
nice!

Reply:
`FutureNCCL` is also mentioned in our docs. Let update those too:

https://github.com/pytorch/pytorch/blob/dee82ef3ea1ed2cae4e891f071adff03cb4a6e0d/torch/csrc/distributed/c10d/init.cpp#L1140-L1144